### PR TITLE
38 courses not in catalogue

### LINKED ISSUE #38: Courses not in Catalogue
As we discussed on Wednesday, some preq strings reference courses that are no longer offered at USask. We would like to add these courses into DB, however, they must have a clear flag. The file 'postDependentCourses.py' has a notInCatalogue list
 which stores all course codes that do not have a matching row entry in the db (and therefore are not in the course catalogue). We would like to add these to the db and create a flag for them. Create the flag as you see fit, however, if you add another column (which you likely will need to), you will have to edit models and all other post/put functions. Big picture:

1. Decide how to flag courses and make this update in all the affected files
2. In post dependency, post all courses not in catalogue to DB with the flag

### PR DESCRIPTION
**Pull request by marmikpatel19**
A new column `not_in_catalogue` is added to CourseTable2023 table. 0 indicates that the course exists; 1 indicates that it does not. This should be checked client-side to render a stub course card for courses that do not exist. 

Examples courses in MySQL db:
image

### CODE REVIEW

File: server/api/migrations/0007_coursetable2023_remove_dependency_course_id_and_more.py
Comment (fraser-mcleod) [inline server/api/migrations/0007_coursetable2023_remove_dependency_course_id_and_more.py:1]: I could be wrong, but I am not sure if migrations should be included in commit

File: server/api/migrations/__pycache__/0007_coursetable2023_remove_dependency_course_id_and_more.cpython-310.pyc
Comment (fraser-mcleod) [inline server/api/migrations/__pycache__/0007_coursetable2023_remove_dependency_course_id_and_more.cpython-310.pyc:1]: Same with the pycache, I don't think they should be in commit

File: server/webscraping/postDependentCourses.py
Comment (fraser-mcleod) [inline server/webscraping/postDependentCourses.py:1]: Looks good!

**Comment by marmikpatel19** [comment]
Looks like [migrations should be kept](https://stackoverflow.com/questions/28035119/should-i-be-adding-the-django-migration-files-in-the-gitignore-file). I'll .gitignore the pycache.

**Pull request merged**